ha/core/event_manager/event_manager.py

```diff
@@ -320,16 +322,10 @@ def subscribe(self, component: SUBSCRIPTION_LIST, events: List[SubscribeEvent])
                 subscription_key = EVENT_MANAGER_KEYS.SUBSCRIPTION_KEY.value.replace("<component_id>", component)
                 if event.functional_types:
                     for func_type in event.functional_types:
-                        resource = event.resource_type + HA_DELIM + func_type       # Example: resource = node>server
-                        self._store_component_key(subscription_key, resource, event.states)
-                        self._store_event_key(resource, event.states, comp=component)
+                        self._store_component_key(subscription_key, event.resource_type, func_type, event.states)
```

Comment:
Please add testing details of calling"subscribe" function with "all" and other keywords ex server, data etc in "functional_types"

Reply:
Updated test result file with subscribe event tests. Also added unittest. Please review.
https://jts.seagate.com/secure/attachment/515266/functional_type_test_results.txt